Dockerfile

```diff
@@ -0,0 +1,46 @@
+iFROM mhart/alpine-node:5.6.0
+
+# Install required dependencies (Alpine Linux packages)
+RUN apk update && \
+  apk add --no-cache \
+    g++ \
+    gcc \
+    git \
+    libev-dev \
+    libevent-dev \
+    libuv-dev \
+    make \
+    openssl-dev \
+    perl \
+    python
+
+# Switch to directory for external dependencies (installed from source)
+WORKDIR /external
+
+# Install (global) NPM packages/dependencies
+RUN npm install -g \
+  node-gyp
+
+# Switch to directory with sources
+WORKDIR /src
+
+# Copy package json and install dependencies
+COPY package.json .
```

Comment:
This tradeoff between modularity and caching. I did not wanted to make it one big cache-liner if you understand what I mean ;-)